doc/barclamp.md

```diff
@@ -0,0 +1,1224 @@
+# Barclamp development guide
+
+This document is an introduction to Barclamp development. It will guide you
+through all the steps neccessary to create a minimal (without UI elements)
+Barclamp for Crowbar. It will then guide you through all the testing, CI and
+review steps until the point where it gets merged into Crowbar.
+
+## What is a Barclamp?
+
+A barclamp is a plugin for Crowbar that configures a particular service or
+group of services. It contains a Chef cookbook with one or more roles to
+perform the actual configuration work and Crowbar UI code to parametrize this
+chef cookbook and orchestrate applying its roles to individual machines.
+
+## Terms and definitions
+
+We will assume throughout this text that your barclamp is will be named
+`mybarclamp`. Please substitute your actual barclamp's name for this
+placeholder when you create a barclamp. The convention for OpenStack barclamps
+is to use the OpenStack component's name, e.g. `nova`.
+
+## Prerequisites
+
+We will assume some things to already be present. Namely, the following:
+
+* A Github fork of the appropriate Barclamp collection. For instance, you will
+  find OpenStack related barclamps in the
+  [crowbar-openstack](https://www.github.com/crowbar/crowbar-openstack)
+  repository and high availability related Barclamps in the
+  [crowbar-ha](https://github.com/crowbar/crowbar-ha) repository. For the sake
+  of readability we will assume you are developing an OpenStack related and
+  forked [crowbar-openstack](https://www.github.com/crowbar/crowbar-openstack)
+  throughout the rest of this document.
+
+* A Crowbar admin node with the Crowbar version you are developing against
```

Comment:
Forgot to respond to this comment yesterday: good point, thanks. I always do this on autopilot so I forgot mentioning it. I added these two to the prerequisites list now.